math/src/fft/fft_inputs.rs

```diff
@@ -172,3 +335,65 @@ pub(super) fn fft_in_place<B, I>(
         }
     }
 }
+
+// TRANSPOSING
+// ================================================================================================
+
+#[cfg(feature = "concurrent")]
+fn transpose_square_stretch<E, I>(matrix: &mut I, size: usize, stretch: usize)
+where
+    E: FieldElement,
+    I: FftInputs<E> + ?Sized,
+{
+    assert_eq!(matrix.len(), size * size * stretch);
+    match stretch {
+        1 => transpose_square_1(matrix, size),
+        2 => transpose_square_2(matrix, size),
+        _ => unimplemented!("only stretch sizes 1 and 2 are supported"),
```

Comment:
Is this a tech-debt? If yes, probably should come with a tracking issue. If no, then probably in `unreachable` instead of `unimplemented`